src/openfl/display/DisplayObject.hx

```diff
@@ -854,6 +854,45 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 	}
 	
 	
+	private function __canvasPushMask (renderSession:RenderSession):Void {
+		
+		if (!cacheAsBitmap) {
+			
+			var context = renderSession.context;
+			var transform = __renderTransform;
+			var pixelRatio = renderSession.pixelRatio;
+			context.setTransform (transform.a * pixelRatio, transform.b, transform.c, transform.d * pixelRatio, transform.tx * pixelRatio, transform.ty * pixelRatio);
+			
+			context.beginPath ();
+			__renderCanvasMask (renderSession);
+			context.clip ();
+			
+		}
+		
+	}
+	
+	
+	private function __canvasPopMask (renderSession:RenderSession):Void {
+		
+		if (cacheAsBitmap) {
+			
+			__isMask = false;
+			__updateCacheBitmap (renderSession, !__worldColorTransform.__isDefault ());
+			__isMask = true;
+			__updateChildren(true);
```

Comment:
I don't like this very much, but also I'm not sure how to deal with this without rewriting everything, argh... Gotta think more about this after fixing other issues.